src/models/link/mod.rs

```diff
@@ -0,0 +1,72 @@
+//! in-toto link
+
+use std::collections::{BTreeMap};
+use std::fmt::{Debug};
+use std::str;
+
+use serde_derive::{Deserialize, Serialize};
+use crate::Result;
+
+pub mod metadata;
+pub use metadata::{LinkMetadata, LinkMetadataBuilder};
+
+use crate::models::{VirtualTargetPath, TargetDescription};
+
+// FIXME, we need to tag a spec
+//const SPEC_VERSION: &str = "0.9-dev";
+
+// FIXME: methods will be relevant for layout expiration
+// fn parse_datetime(ts: &str) -> Result<DateTime<Utc>> {
+//     Utc.datetime_from_str(ts, "%FT%TZ")
+//         .map_err(|e| Error::Encoding(format!("Can't parse DateTime: {:?}", e)))
+// }
+//
+// fn format_datetime(ts: &DateTime<Utc>) -> String {
+//     format!(
+//         "{:04}-{:02}-{:02}T{:02}:{:02}:{:02}Z",
+//         ts.year(),
+//         ts.month(),
+//         ts.day(),
+//         ts.hour(),
+//         ts.minute(),
+//         ts.second()
+//     )
+// }
+
+pub const FILENAME_FORMAT: &str = "{step_name}.{keyid:.8}.link";
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Link {
+    // Why is the type named as typ?
+    #[serde(rename = "_type")]
```

Comment:
btw, I think this is missing the actual type field itself. The original code had `typ`.

Reply:
I actually removed the field typ because it was related to TUF role, and doesn't say anything aside from "Link" (or "layout" I'm guessing once we implement layout).

Should it not have been deleted?

Reply:
Sure, I think it's okay to use "link" directly since we have fewer instances. But we still need to account for that field? plus, the serde statement now applies to `name` if I'm not mistaken?